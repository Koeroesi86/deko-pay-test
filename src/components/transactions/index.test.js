import React from 'react';
import { shallow, mount } from 'enzyme';
import axios from 'axios';
import Transactions from './index';
import { act } from "@testing-library/react";

const mockTransactions = {
  "name": "Rohan LLC Shoes",
  "merchant_id": "A1201",
  "pricing": {
    "subsidy": 9,
    "discount_subsidy": 6,
    "discount_cutoff": 600
  },
  "transactions": [
    {
      "description": "Granite Keyboard",
      "date": "2019-02-20T15:06:32.153Z",
      "price": 1675
    },
    {
      "description": "Generic Wooden Bacon",
      "date": "2019-03-08T17:58:17.636Z",
      "price": 2540
    },
    {
      "description": "Pants",
      "date": "2019-03-11T11:48:04.699Z",
      "price": 1558
    }
  ]
};

jest.mock('axios');
axios.get.mockImplementation(() => Promise.resolve({ data: mockTransactions }));

describe('The Transactions component', () => {
  it('renders', () => {
    const result = shallow(<Transactions />);
    expect(result).toMatchSnapshot();
  });



  it('mounts', async () => {
    const merchant = 'A1201';
    let result;
    await act(async () => {
      result = mount(<Transactions merchant={merchant} />);
    });
    result.update();

    expect(axios.get).toHaveBeenCalledWith(`http://interview.dekopay.com.s3.eu-west-2.amazonaws.com/merchants/${merchant}.json`);
    expect(result).toMatchSnapshot();
  });
});
