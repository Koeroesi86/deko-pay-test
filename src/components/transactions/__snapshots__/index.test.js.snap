// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Transactions component mounts 1`] = `
<div
  className="Transactions"
>
  <div
    className="headerWrapper"
  >
    <div
      className="meta"
    >
      <div
        className="header"
      >
        Transactions
      </div>
      <div
        className="merchantName"
      >
        Rohan LLC Shoes
      </div>
      <div
        className="pricing"
      >
        <div
          className="item"
        >
          <div
            className="title"
          >
            Count
          </div>
          <div
            className="value"
          >
            3
          </div>
        </div>
        <div
          className="item"
        >
          <div
            className="title"
          >
            Total
          </div>
          <div
            className="value"
          >
            £
            5773.00
          </div>
        </div>
        <div
          className="item"
        >
          <div
            className="title"
          >
            Subsidy
          </div>
          <div
            className="value"
          >
            £
            346.38
          </div>
        </div>
      </div>
    </div>
    <svg
      className="pie"
      viewBox="0 0 20 20"
    >
      <circle
        className="base"
        cx="10"
        cy="10"
        r="10"
      />
      <circle
        className="slice"
        cx="10"
        cy="10"
        r="5"
        strokeDasharray="calc(21.599999999999998 * 31.4 / 100) 31.4"
        strokeWidth="10"
        transform="rotate(-90) translate(-20)"
      />
    </svg>
  </div>
  <div
    className="listing"
  >
    <div
      className="row header"
    >
      <div
        className="column description"
      >
        Description
      </div>
      <div
        className="column date"
      >
        Date
      </div>
      <div
        className="column price"
      >
        Price
      </div>
      <div
        className="column subsidy"
      >
        Subsidy
      </div>
    </div>
    <div
      className="row transaction"
      key="transaction-0"
    >
      <div
        className="column description"
      >
        Granite Keyboard
      </div>
      <div
        className="column date"
      >
        2019-02-20 15:06:32
      </div>
      <div
        className="column price"
      >
        £
        1675.00
      </div>
      <div
        className="column subsidy"
      >
        £
        100.50
      </div>
    </div>
    <div
      className="row transaction"
      key="transaction-1"
    >
      <div
        className="column description"
      >
        Generic Wooden Bacon
      </div>
      <div
        className="column date"
      >
        2019-03-08 17:58:17
      </div>
      <div
        className="column price"
      >
        £
        2540.00
      </div>
      <div
        className="column subsidy"
      >
        £
        152.40
      </div>
    </div>
    <div
      className="row transaction"
      key="transaction-2"
    >
      <div
        className="column description"
      >
        Pants
      </div>
      <div
        className="column date"
      >
        2019-03-11 11:48:04
      </div>
      <div
        className="column price"
      >
        £
        1558.00
      </div>
      <div
        className="column subsidy"
      >
        £
        93.48
      </div>
    </div>
  </div>
</div>
`;

exports[`The Transactions component renders 1`] = `
<div
  className="Transactions"
/>
`;
